### PR TITLE
Upgrade to Realm Core v13.20.0

### LINKED ISSUE #6077: Upgrade to Realm Core v13.20.0

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

This closes #6077 

## ☑️ ToDos

* [x] 📝 Changelog entry
* [ ] 📝 `Compatibility` label is updated or copied from previous entry
* [ ] 📝 Update `COMPATIBILITY.md`
* [ ] 🚦 Tests
* [ ] 📦 Updated internal package version in consuming `package.json`s (if updating internal packages)
* [ ] 📱 Check the React Native/other sample apps work if necessary
* [ ] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

**Review by takameyer — APPROVED** [review]
LGTM :shipit:

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,9 +4,13 @@
 * `Realm.Sync.reconnect(app)` has been deprecated and will be removed in the next major version. You can use `Realm.Sync.Session.reconnect()` instead.
```
Comment (elle-j) [inline CHANGELOG.md:4]:
```suggestion
* `Realm.App.Sync.reconnect(app)` has been deprecated and will be removed in the next major version. You can use `Realm.App.Sync.Session.reconnect()` instead.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,9 +4,13 @@
...
 
 ### Enhancements
-* Added `Realm.Sync.Session.reconnect()` to help force a reconnection to Atlas Device Sync. ([#6123](https://github.com/realm/realm-js/issues/6123))
+* Full text search supports searching for prefix only e.g., `description TEXT 'alex*'`. ([realm/realm-core#6860](https://github.com/realm/realm-core/issues/6860))
+* Unknown protocol errors received from the Altas Device Sync server will no longer cause the app to crash if a valid error action is also received. Unknown error actions will cause device sync to fail with an error via the sync error handler. ([realm/realm-core#6885](https://github.com/realm/realm-core/pull/6885))
+* Allow arguments to Realm Query Language to be a string representation of a geospatial object for `GEOWITHIN` queries. ([realm/realm-core#6934](https://github.com/realm/realm-core/issues/6934))
 
 ### Fixed
+* Crash when querying the size of a Object property through a link chain. ([realm/realm-core#6915](https://github.com/realm/realm-core/issues/6915), since v12.0.0)
+* Added `Realm.Sync.Session.reconnect()` to help force a reconnection to Atlas Device Sync. ([#6123](https://github.com/realm/realm-js/issues/6123))
```
Comment (elle-j) [inline CHANGELOG.md:13]:
```suggestion
* Added `Realm.App.Sync.Session.reconnect()` to help force a reconnection to Atlas Device Sync. ([#6123](https://github.com/realm/realm-js/issues/6123))
```

File: integration-tests/tests/src/tests/sync/sync-session.ts [hunk 3]
```diff
@@ -278,7 +278,7 @@ describe("SessionTest", () => {
           const realm = new Realm(config);
           const session = realm.syncSession;
           //@ts-expect-error using internal method.
-          session._simulateError(123, "simulated error", "realm::sync::ProtocolError", false);
+          session._simulateError(123, "simulated error", "realm::sync::ProtocolError", false); // this will lead to error code 1032
```
Comment (elle-j) [inline integration-tests/tests/src/tests/sync/sync-session.ts:281]: Should the 1st argument here also be changed to `1032` as on line 267?

**Review by elle-j — APPROVED** [review]
👏 !

File: integration-tests/tests/src/tests/sync/sync-session.ts (hunk 3 above)
Comment (kneth) [reply integration-tests/tests/src/tests/sync/sync-session.ts:281]: Good point. 123 isn't used by core

**Pull request merged**